### PR TITLE
Secure OCF

### PR DESCRIPTION
**Pull request by mmichal10**

### CODE REVIEW

File: tests/functional/tests/utils/random.py [hunk 1]
```diff
@@ -66,22 +66,23 @@ def __next__(self):
 
 
 class RandomStringGenerator:
-    def __init__(self, len_range=Range(0, 20), count=700):
+    def __init__(self, len_range=Range(0, 20), count=700, chars_list=[]):
```
Comment (arutk) [inline tests/functional/tests/utils/random.py:69]: maybe rename to "extra_chars"?

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 2]
```diff
@@ -466,3 +467,213 @@ def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
...
+        cache_mode=cm,
+        cache_line_size=cls,
+    )
+
+    # Set invalid cache mode and check if failed
+    for i in RandomGenerator(DefaultRanges.INT):
+        if i in CacheMode + [CACHE_MODE_NONE]:
+            continue
+        with pytest.raises(OcfError, match="Error adding partition to cache"):
+            cache.configure_partition(
```
Comment (arutk) [inline tests/functional/tests/security/test_management_fuzzy.py:612]: wait, we can configure cache mode per IO class :O ?

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 3]
```diff
@@ -466,3 +467,213 @@ def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
...
+@pytest.mark.security
+def test_neg_set_ioclass_name(pyocf_ctx):
+    """
+    Test whether it is possible to add ioclass with invaild name
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
+    :return:w
+    """
+    invalid_chars_codes = list(range(0, ord(" "))) + list(range(ord("~")+1, 255))
```
Comment (arutk) [inline tests/functional/tests/security/test_management_fuzzy.py:632]: I think you mean non "printable" characters - there is a list for that string class

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 4]
```diff
@@ -466,3 +467,213 @@ def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
...
+    cache_device = Volume(S.from_MiB(30))
+    cache = Cache.start_on_device(
+        cache_device,
+        cache_mode=CacheMode.WT,
+        cache_line_size=CacheLineSize.LINE_4KiB,
+    )
+
+    # Set invalid name and check if failed
+    for name in RandomStringGenerator(len_range=Range(0, 4096), count=10000):
+        if not any(c for c in invalid_chars if c in name) or len(name) <= 1024:
```
Comment (arutk) [inline tests/functional/tests/security/test_management_fuzzy.py:680]: no need to have invalid chars here, sufficient length is enough

File: inc/ocf_def.h [hunk 5]
```diff
@@ -176,6 +176,8 @@ typedef enum {
 } ocf_cache_mode_t;
 
 #define OCF_SEQ_CUTOFF_MAX_STREAMS 256
+#define OCF_SEQ_CUTOFF_MIN_THRESHOLD 1
+#define OCF_SEQ_CUTOFF_MAX_THRESHOLD 4194181
```
Comment (arutk) [inline inc/ocf_def.h:180]: why not UIN32_T?

File: src/mngt/ocf_mngt_io_class.c [hunk 6]
```diff
@@ -256,6 +256,26 @@ static int _ocf_mngt_io_class_validate_cfg(ocf_cache_t cache,
...
+				"Invalid value of the partition max size\n");
+		return -OCF_ERR_INVAL;
+	}
+
+	if (cfg->min_size < PARTITION_SIZE_MIN ||
+			cfg->min_size > PARTITION_SIZE_MAX) {
+		ocf_cache_log(cache, log_info,
+				"Invalid value of the partition min size\n");
+		return -OCF_ERR_INVAL;
+	}
```
Comment (arutk) [inline src/mngt/ocf_mngt_io_class.c:271]: Is the "min size" being validated? I think no, I would just remove this from IO class config.

**Review by arutk — CHANGES_REQUESTED** [review]

File: inc/ocf_def.h (hunk 5 above)
Comment (mmichal10) [reply inc/ocf_def.h:180]: I've seen this value in requirements: https://github.com/Open-CAS/open-cas-linux/pull/699/files

File: src/mngt/ocf_mngt_io_class.c (hunk 6 above)
Comment (mmichal10) [reply src/mngt/ocf_mngt_io_class.c]: Ok, I removed it from the public API

**Review by arutk — APPROVED** [review]

**Review by arutk — APPROVED** [review]

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 7]
```diff
@@ -454,3 +456,143 @@ def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
...
+            print("\n" + i)
+
+
+@pytest.mark.security
+def test_neg_set_ioclass_name(pyocf_ctx):
+    """
+    Test whether it is possible to add ioclass with invaild name
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
```
Comment (rafalste) [inline tests/functional/tests/security/test_management_fuzzy.py:564]: No such params here.

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 8]
```diff
@@ -454,3 +456,143 @@ def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
...
+            print(f"\n{name}")
+
+
+@pytest.mark.security
+def test_neg_set_ioclass_name_len(pyocf_ctx):
+    """
+    Test whether it is possible to add ioclass with too long name
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
```
Comment (rafalste) [inline tests/functional/tests/security/test_management_fuzzy.py:593]: No such params here.

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 9]
```diff
@@ -454,3 +456,143 @@ def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
...
+    # Start cache device
+    cache_device = Volume(S.from_MiB(30))
+    cache = Cache.start_on_device(
+        cache_device, cache_mode=CacheMode.WT, cache_line_size=CacheLineSize.LINE_4KiB
+    )
+
+    # Set invalid name and check if failed
+    for name in RandomStringGenerator(len_range=Range(0, 4096), count=10000):
+        if len(name) <= 1024:
+            continue
```
Comment (rafalste) [inline tests/functional/tests/security/test_management_fuzzy.py:606]: Maybe just use `Range(1025, 4096)`? Or even some `max_name_len` variable?

**Review by rafalste — APPROVED** [review]

**Review by arutk — APPROVED** [review]

**Pull request merged**